opus/application/static_media/js/browse.js

```diff
@@ -796,6 +731,176 @@ var o_browse = {
         }
     },
 
+    realignDOMAndGetStartObsAndScrollbarObsNum: function(selector, tab, viewNamespace, galleryBoundingRect,
+                                                         browserResized) {
+        /**
+         * get the current startObs (obsNum, for slider number) and the
+         * top item's obsNum in table view, this item will also be at the
+         * top row in gallery view (currentScrollObsNum, for setting
+         * scrollbar location).
+         */
+
+        // this will get the top left obsNum for gallery view or the top obsNum for table view
+        let firstCachedObs = $(selector).first().data("obs");
+        if (browserResized) {
+            o_browse.deleteCachedObs(tab, viewNamespace, galleryBoundingRect, firstCachedObs);
+        }
+
+        let firstCachedObsTop = $(selector).first().offset().top;
+        let alignedCachedFirstObs = (o_utils.floor((firstCachedObs - 1)/galleryBoundingRect.x) *
+                                     galleryBoundingRect.x + 1);
+
+        // For gallery view, the topBoxBoundary is the top of .gallery-contents
+        // For table view, we will set the topBoxBoundary to be the bottom of thead
+        // (account for height of thead)
+        let topBoxBoundary = (o_browse.isGalleryView() ?
+                              $(`${tab} .gallery-contents`).offset().top :
```

Comment:
might want to consider creating a variable for the gallery-contents   .top value, as it seems to be used in multiple places

Reply:
I think this has not been resolved.